src/manager/evm.rs

```diff
@@ -26,127 +30,213 @@ pub type MiddlewareImpl = SignerMiddleware<Provider<Http>, Wallet<SigningKey>>;
 
 // Create type bindings for the IPC Solidity contracts
 abigen!(Gateway, "contracts/Gateway.json");
-abigen!(SubnetActor, "contracts/SubnetActor.json");
+abigen!(SubnetContract, "contracts/SubnetActor.json");
+abigen!(IPCRegistry, "contracts/IPCRegistry.json");
 
 pub struct EthSubnetManager<M: Middleware> {
     eth_client: Arc<M>,
+    gateway_contract: Gateway<Arc<M>>,
+    registry_contract: IPCRegistry<Arc<M>>,
 }
 
 #[async_trait]
-impl<M: Middleware + Send + Sync> SubnetManager for EthSubnetManager<M> {
-    async fn create_subnet(&self, from: Address, params: ConstructParams) -> Result<Address> {
-        todo!()
+impl<M: Middleware + Send + Sync + 'static> SubnetManager for EthSubnetManager<M> {
+    async fn create_subnet(&self, _from: Address, params: ConstructParams) -> Result<Address> {
+        let min_validator_stake = params
+            .min_validator_stake
+            .atto()
+            .to_u64()
+            .ok_or_else(|| anyhow!("invalid min validator stake"))?;
+
+        let params = IsubnetActorConstructorParams {
+            // TODO: replace this with parent
+            parent_id: ipc_registry::SubnetID {
+                route: agent_subnet_to_evm_addresses(&params.parent)?,
+            },
+            name: params.name,
+            ipc_gateway_addr: (*self.gateway_contract).address(),
+            consensus: params.consensus as u64 as u8,
+            min_activation_collateral: ethers::types::U256::from(min_validator_stake as u128),
+            min_validators: params.min_validators,
+            bottom_up_check_period: params.bottomup_check_period as u64,
+            top_down_check_period: params.topdown_check_period as u64,
+            // TODO: update this variable properly
+            majority_percentage: 50,
+            genesis: ethers::types::Bytes::default(),
+        };
+        log::info!("creating subnet: {params:?}");
+
+        let call = self.registry_contract.new_subnet_actor(params);
+        let pending_tx = call.send().await?;
+        let receipt = pending_tx.retries(10).await?;
```

Comment:
Let's add a comment explaining that we need to use retries per this issue: https://github.com/filecoin-project/community/discussions/638